pkg/k8s/chart/releaseinstance.go

```diff
@@ -0,0 +1,75 @@
+package chart
+
+import (
+	"errors"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/imdario/mergo"
+)
+
+func NewReleaseInstance(name, namespace string, configuration map[string]interface{}) *ReleaseInstance {
+	return &ReleaseInstance{
+		Name:          name,
+		Namespace:     namespace,
+		Configuration: configuration,
+	}
+}
+
+type ReleaseInstance struct {
+	Name              string
+	Namespace         string
+	Configuration     map[string]interface{}
+	RenderedManifests ManifestResources
```

Comment:
Why is this information exported?
IMO, it is dangerous to allow those fields to be modifiable by any entity that has access to the release instance.

Reply:
I only used the struct to pass data among methods. But yes, valid point. I will do it in later PR, as it may fail some things in the parent PR. I have added a task [here](https://github.com/kyma-project/nats-manager/issues/8) for it.